### PR TITLE
add reddit link to description

### LINKED ISSUE #4: add reddit link to rss feed
just thought it might be worth adding a link to the origonal reddit post to the rss feed

think the easiest way to do this is to stick 
```
description += `<p><a href="${comic.link}">Origonal Post</a></p>\`;
```
after the loop that adds the images

i think at the bottom is better since that is where people will be when they want to go to the reddit post (ie, to read comments)

i dont think it should be the main link since that is used to link to spesific points in a series as well and this site is a better interface for looking at a series that reddit.

### PR DESCRIPTION
**Pull request by TomW1605**
closes #4

### CODE REVIEW

File: front/app/feed.xml/route.tsx [hunk 1]
```diff
@@ -39,16 +40,24 @@ export async function GET() {
...
+              alt={image.ocr}
+              width={image.width}
+              height={image.height}
+            />
+          );
+        })}
+        <p>
+          <a href={comic.link}>Original Post</a>
+        </p>
+      </>,
```
Comment (HakierGrzonzo) [inline front/app/feed.xml/route.tsx:59]: Could we extract this logic to a separate react component? Something called  RSSdescription in the components folder would be nice.

Comment (TomW1605) [reply front/app/feed.xml/route.tsx:59]: i did try that but it seemed like an unnesaserry complication since the route file would have to be renamed anyway to use the component. happy to switch to that though if you think it makes sense

Comment (HakierGrzonzo) [reply front/app/feed.xml/route.tsx:59]: @TomW1605 I meant moving this JSX code (The `<p>bracket things</p>`) to a separate file and then passing some `<RSSdescription metadata={metadata}/>` to the `renderToString` in the `route.tsx`.

This way we will have one place with all the RSS logic and another with _the html for description_. When changing the description we don't need to worry about all the RSS feed things (e.g. that we render it to string etc.), and when we change the RSS logic we don't need to worry about how the description looks like.

Comment (TomW1605) [reply front/app/feed.xml/route.tsx:59]: ok, yeah, that works. i need to pass through the comic and the images not just the metadata otherwise we end up redoing all the sorting process as well but we end up with this in the route file
```js
let description = renderToString(
  <>
    <RSSdescription comic={comic} images={metadata.images} />
  </>,
);
```

**Review by HakierGrzonzo — COMMENTED** [review]
LGTM, will test if it works and then we can merge it

**Comment by HakierGrzonzo** [comment]
Found one bug due to missing `key` in the loop to create ``, but I fixed it. Will make github actions run `npm lint` automatically in the future.

As for the feature, it nicely renders in my test viewer:
image

**Review by HakierGrzonzo — APPROVED** [review]

**Comment by HakierGrzonzo** [comment]
@TomW1605 Thanks for the contribution!

I have setup some CD pipelines, so the changes should be visible on the site auto-magically when the cache refreshes (so up to an hour).

**Comment by TomW1605** [comment]
looks good, thanks for all your work

> Found one bug due to missing `key` in the loop to create ``

that isnt a standard attribute for the img tag, is that a react/node thing?

**Comment by HakierGrzonzo** [comment]
It is a react thing, see more about it [in the docs](https://react.dev/learn/rendering-lists#keeping-list-items-in-order-with-key).

TL;DR is that react needs to know which `` tag corresponds to which element of the list, so in case that the list changes (it never happens here as we render it to string), it knows which html `` it should change.
